x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/literal/IntervalUtilsTests.java

```diff
@@ -0,0 +1,223 @@
+import static org.elasticsearch.xpack.sql.tree.Location.EMPTY;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_DAY;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_DAY_TO_HOUR;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_DAY_TO_MINUTE;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_DAY_TO_SECOND;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_HOUR;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_HOUR_TO_MINUTE;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_HOUR_TO_SECOND;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_MINUTE;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_MINUTE_TO_SECOND;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_MONTH;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_SECOND;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_YEAR;
+import static org.elasticsearch.xpack.sql.type.DataType.INTERVAL_YEAR_TO_MONTH;
+
+public class IntervalUtilsTests extends ESTestCase {
+
+    public void testYearInterval() throws Exception {
+        String randomSign = randomSign();
+        int random = randomNonNegativeInt();
+        TemporalAmount amount = parseInterval(EMPTY, randomSign + random, INTERVAL_YEAR);
+        assertEquals(maybeNegate(randomSign, Period.ofYears(random)), amount);
+    }
+
+    public void testMonthInterval() throws Exception {
+        String randomSign = randomSign();
+        int random = randomNonNegativeInt();
+        TemporalAmount amount = parseInterval(EMPTY, randomSign + random, INTERVAL_MONTH);
+        assertEquals(maybeNegate(randomSign, Period.ofMonths(random)), amount);
+    }
+
+    public void testDayInterval() throws Exception {
+        String randomSign = randomSign();
+        int random = randomNonNegativeInt();
+        TemporalAmount amount = parseInterval(EMPTY, randomSign + random, INTERVAL_DAY);
+        assertEquals(maybeNegate(randomSign, Duration.ofDays(random)), amount);
+    }
+
+    public void testHourInterval() throws Exception {
+        String randomSign = randomSign();
+        int random = randomNonNegativeInt();
+        TemporalAmount amount = parseInterval(EMPTY, randomSign + random, INTERVAL_HOUR);
+        assertEquals(maybeNegate(randomSign, Duration.ofHours(random)), amount);
+    }
+
+    public void testMinuteInterval() throws Exception {
+        String randomSign = randomSign();
+        int random = randomNonNegativeInt();
+        TemporalAmount amount = parseInterval(EMPTY, randomSign + random, INTERVAL_MINUTE);
+        assertEquals(maybeNegate(randomSign, Duration.ofMinutes(random)), amount);
+    }
+
+    public void testSecondInterval() throws Exception {
+        String randomSign = randomSign();
+        int randomSeconds = randomNonNegativeInt();
+        int randomMillis = randomInt(999999999);
```

Comment:
Isn't this a too large interval for random tests, not to do add some tests for scenarios like 0 millis?

Reply:
What I meant is that for a random number with that maximum value there will be rare situations where something like `0` or `10` milliseconds will be used in a test. My suggestion was to specifically add a test for a very small value of millis and, also, a very large one, respectively.

Reply:
Added a random which picks the limits (0 or 9999999) or a random of the interval.